util/patch/patch.go

```diff
@@ -301,3 +302,15 @@ func (h *Helper) calculateChanges(after runtime.Object) (map[string]bool, error)
 	}
 	return res, nil
 }
+
+// PatchFinalizer adds target finalizer if not exist
+func (h *Helper) PatchFinalizer(ctx context.Context, obj controllerutil.Object, finalizer string) (patched bool, reterr error) {
+	if !controllerutil.ContainsFinalizer(obj, finalizer) {
+		controllerutil.AddFinalizer(obj, finalizer)
+		if err := h.Patch(ctx, obj); err != nil {
```

Comment:
Just a note that this Patch is technically able to patch other things in the object as well, if they have changed. For example if the finalizer doesn't exist at this point and we've changed status before calling this method, the status object will be patched as well.

Having said that, do you think we still need this method, or can we reuse the normal helper directly?